### PR TITLE
Redundant titles

### PR DESCRIPTION
**Pull request by brucemiller**
This PR avoids the accidental duplication in the `<title>` element;  keeps the `@title` on html anchors avoid repeating the contents, to improve accessibility. But also provides for `link` elements to get the full, out-of-context, text.

Fixes: #2162
Fixes: #2169

### CODE REVIEW

File: lib/LaTeXML/resources/RelaxNG/LaTeXML-inline.rnc [hunk 1]
```diff
@@ -175,10 +175,15 @@ ref_attributes =
...
   attribute show { text }?,
 
+  ## gives a description of the target, not repeating the content,
+  ## used for accessibility or a tooltip in HTML.
+  ## Typically filled in by postprocessor.
+  attribute title { text }?,
+
   ## gives a longer form description of the target,
-  ## this would typically appear as a tooltip in HTML.
+  ## usedful when the link appears outside its original context, eg in navigation.
```
Comment (dginev) [inline lib/LaTeXML/resources/RelaxNG/LaTeXML-inline.rnc:184]: `usedful` typo

**Review by dginev — APPROVED** [review]
Looks good!

**Comment by brucemiller** [comment]
I meant to also ping @rzach to see if this worked out for him.  And also a question about appropriate style for combining title components.

In a hierarchically structured website like DLMF it's *very* helpful for `@title` to show the context of each link; so for https://dlmf.nist.gov/1.7.ii you get  "\S1.7(ii) Integrals IN \S1.7 Inequalities IN Ch. 1 Algebraic and Analytic Methods".  It seemed to me that inserting the english word "IN" was potentially confusing (and weird, if the site is not english!), so I arbitrarily chose a triangle (U+2023) as the separator.  It looks plausible to my eye, but maybe not so accessible?

That question is even more acute since we're now omitting the "\S1.7(ii) Integrals" when it's already present in the link's content. Having `@title` start with "\S1.7 Inequalities" is misleading; starting with the triangle is very weird; So, I fell back to prefixing these cases with "In ".  So, the above link would get `@title` starting "In \S1.7 Inequalities...".

So, feedback or suggestions are welcome.

**Comment by brucemiller** [comment]
I don't quite understand what you're suggesting; both the spec (somewhere) and the WAVE tool are saying that `@title` is intended for accessibility (as well as tooltip elsewhere in the same spec).

**Comment by dginev** [comment]
> I don't quite understand what you're suggesting; both the spec (somewhere) and the WAVE tool are saying that `@title` is intended for accessibility (as well as tooltip elsewhere in the same spec).

WAI-ARIA explicitly calls out `@title` and `@placeholder` as fallback attributes and recommends against reliance on them. They recommend the `aria-label` for the text which is always read:
https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#naming_rule_avoid_fallback

So my suggestion was to optimize `@title` for tooltip use, and set `aria-label` for accessible vocalization, when that is needed. And I am OK with holding off on that for now, as `<a>` links have decent vocalization in our current HTML output.

**Review by dginev — APPROVED** [review]
Looks good!

**Comment by rzach** [comment]
I haven't been able to check what an actual screen reader (JAWS or NVDA) does but given what @dginev says in https://github.com/brucemiller/LaTeXML/pull/2207#issuecomment-1700242563 I fully expect that they don't exhibit the weird behavior I saw in ChromeVox. So I'm happy with it.

**Comment by dginev** [comment]
Btw @rzach when you mentioned in the issue:
> that's what the screen reader (ChromeVox) is doing: it assumes that what's in the title attribute is important and so reads it out, but of course also reads out the link text. 

That felt like some sort of strange bug on ChromeVox's part. I don't have it available so can't verify (although Chromebooks are cheap, so maybe I should grab one). When I tried Chrome's Screen Reader extension (available on any OS), it didn't read out the `@title` attribute on your site or the DLMF. Which is reasonable. I would have expected a choice of *either* reading the `<a>` content, *or* reading the `@title` attribute. Reading both could be more frustrating than helpful.

**Comment by rzach** [comment]
Yes several people who know about a11y and SRs subsequently to me filing the bug told me that the ChromeVox extension is not good... but I'm on Linux so I haven't yet been able to test it with a SR that is actually good. Even the built-in SR on Chromebooks might be much better.

**Comment by brucemiller** [comment]
I was also going by the warnings that the WAVE extension gives (and trying to minimize them).

**Pull request merged**